### ISSUE TITLE
does cflags work in compilers.yaml

### ISSUE THREAD

**Issue by glennpj** [LLNL/spack#1512]
I have the following in compilers.yaml
```
compilers:
- compiler:
    modules: []
    operating_system: centos7
    paths:
      cc: /usr/bin/gcc
      cxx: /usr/bin/g++
      f77: /usr/bin/gfortran
      fc: /usr/bin/gfortran
    spec: gcc@4.8.5
- compiler:
    modules: []
    operating_system: centos7
    paths:
      cc: /home/gjohnson/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-6.1.0-u2wxtnirnnygyyozth2ylds56sbdnm5a/bin/gcc
      cxx: /home/gjohnson/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-6.1.0-u2wxtnirnnygyyozth2ylds56sbdnm5a/bin/g++
      f77: /home/gjohnson/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-6.1.0-u2wxtnirnnygyyozth2ylds56sbdnm5a/bin/gfortran
      fc: /home/gjohnson/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-6.1.0-u2wxtnirnnygyyozth2ylds56sbdnm5a/bin/gfortran
      cflags: -march=native
      fflags: -march=native
      cxxflags: -march=native
    spec: gcc@6.1.0
```
However, I do not see any evidence that things built with gcc@6.1.0 are using the flags values.
```
spec:
- ncurses:
    arch:
      platform: linux
      platform_os: centos7
      target: x86_64
    compiler:
      name: gcc
      version: 6.1.0
    hash: fmqjxbzwxst76zjvhxcxs227jliyidnl
    namespace: builtin
    parameters:
      cflags: []
      cppflags: []
      cxxflags: []
      fflags: []
      ldflags: []
      ldlibs: []
    version: '6.0'
```

**Comment by xjrc** [comment]
This is a bug that I was experiencing as well.  I've opened a pull request to attempt to fix the issue (see #1532).

**Comment by glennpj** [comment]
@xjrc Thanks, I will try it out as soon as I can.

**Comment by xjrc** [comment]
Fixed with merge of #1532.
